File: easy/Trick-Or-Treat/Trick Or Treat.java
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            ShowAmount(line);
        }
    }
    
    static void ShowAmount(String line){
        String[] kids = line.split(",");
        int result = 0;
        int totalkids = 0;
        for(String kid : kids){
            int amt = Integer.valueOf(kid.substring(kid.indexOf(":")+2));
            if (kid.indexOf("V")>=0){
                totalkids += amt;
                result += 3*amt;
            } else if (kid.indexOf("Z")>=0){
                totalkids += amt;
                result += 4*amt;
            } else if (kid.indexOf("W")>=0){
                totalkids += amt;
                result += 5*amt;
            } else result *= amt;
        }
        System.out.println(result/totalkids);
    }
}

File: easy/Trick-Or-Treat/Trick Or Treat.cs
using System;
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            ShowAmount(line);
        }
    }
    
    static void ShowAmount(string line){
        string[] kids = line.Split(',');
        int result = 0;
        int totalkids = 0;
        foreach(string kid in kids){
            int amt = Convert.ToInt32(kid.Substring(kid.IndexOf(':')+2));
            if (kid.IndexOf('V')>=0){
                totalkids += amt;
                result += 3*amt;
            } else if (kid.IndexOf('Z')>=0){
                totalkids += amt;
                result += 4*amt;
            } else if (kid.IndexOf('W')>=0){
                totalkids += amt;
                result += 5*amt;
            } else result *= amt;
        }
        Console.WriteLine(result/totalkids);
    }
}
